### ISSUE TITLE
RM551E-GL Openwrt

### ISSUE THREAD

**Issue by SpeedxThunder001** [iamromulan/RM551E-GL#7]
Is there any guide to make RM551E-GL to work with openwrt or GL.inet Flint 2 MT-6000 specifically?

**Comment by SpeedxThunder001** [comment]
i am using m.2 ro usb adapter

**Comment by SpeedxThunder001** [comment]
the thing is in mt6000 when its detected as mbim/qcm it provides all the options and config on the router itself but in ecm mode i am only able to use it as tethering mode and everything is handled by the module so i am not able to view and info or change and settings

**Comment by iamromulan** [comment]
I haven't tried to use it in modes other than ECM yet but I'll give that a try when I get home later.

You can install a management interface on the modem itself as the modem does run OpenWRT.

You'll be able to access it through the gateway address.

https://youtu.be/dh7dbEyHwiY?si=Va9qgxIvVP2MOSnu

**Comment by SpeedxThunder001** [comment]
does the inbuilt openwrt also work in rndis mode in windows?

**Comment by iamromulan** [comment]
ECM and RNDIS should be super similar behavior. Yep 👍

**Comment by SpeedxThunder001** [comment]
but windows doesnt recognise the drivers even after installing everything when in rndis mode 

do you have any idea about that?

**Comment by d5aqoep** [comment]
Got access to OpenWRT using your excellent youtube video. I Installed Tailscale on it too and could get access to OpenWRT web gui from some random pc with tailscale installed and authenticated. However, the Terminal is not visible when connected through tailscale.

**Comment by iamromulan** [comment]
> but windows doesnt recognise the drivers even after installing everything when in rndis mode 
> 
> do you have any idea about that?

I don't use RNDIS mode on windows.

I use ECM on windows.

Install the NDIS driver and ECM driver from this repo or QuecDeploy only. Uninstall all other Quectel drivers.

Once done make sure the modem is in usbnet 1 and go to device manager. Right click on the ECM device with the missing driver and click update driver. Choose from list. Network Adapters, Quectel, ECM adapter.

**Comment by iamromulan** [comment]
> Got access to OpenWRT using your excellent youtube video. I Installed Tailscale on it too and could get access to OpenWRT web gui from some random pc with tailscale installed and authenticated. However, the Terminal errors out when connected through tailscale.

Has to do with the ttyd settings on what interface it will listen for requests on.

When you go to terminal in Luci go to config and change the interface to unspecified or tailscale to get it to work over tailscale.

**Comment by d5aqoep** [comment]
> > Got access to OpenWRT using your excellent youtube video. I Installed Tailscale on it too and could get access to OpenWRT web gui from some random pc with tailscale installed and authenticated. However, the Terminal errors out when connected through tailscale.
> 
> Has to do with the ttyd settings on what interface it will listen for requests on.
> 
> When you go to terminal in Luci go to config and change the interface to unspecified or tailscale to get it to work over tailscale.

Can you guide me how to do that? I dont want to bungle it up and lose access to everything.

**Comment by borfoo3** [comment]
Hello
Running Openwrt on the RM551 and 5g2phy I am noticing almost double the latency and ~30% speed drop vs the original device software.
Any tips or ideas how to improve this? I don't see an option for 'software offloading'
As a plan B - is it easy to go back to the device original software?

**Comment by iamromulan** [comment]
> > > Got access to OpenWRT using your excellent youtube video. I Installed Tailscale on it too and could get access to OpenWRT web gui from some random pc with tailscale installed and authenticated. However, the Terminal errors out when connected through tailscale.
> > 
> > Has to do with the ttyd settings on what interface it will listen for requests on.
> > 
> > When you go to terminal in Luci go to config and change the interface to unspecified or tailscale to get it to work over tailscale.
> 
> Just did. Thanks. Is it possible to add both LAN and Tailscale interfaces? 
> I Added 2nd ttyd instance but only 1st instance works and I can only get either LAN or Tailscale to work. Not both together.

It's on the to do list to add something to the tailscale init that will create an alias interface with LAN + Tailscale for firewall rule purposes. Should be able to select it on the ttyd settings in Luci after I do that.

**Comment by iamromulan** [comment]
> Hello
> Running Openwrt on the RM551 and 5g2phy I am noticing almost double the latency and ~30% speed drop vs the original device software.
> Any tips or ideas how to improve this? I don't see an option for 'software offloading'
> As a plan B - is it easy to go back to the device original software?

Original device software? I'm not sure what you mean. The modem firmware is all stock and we tiptoe around not modifying the stock files as much as possible.

**Comment by iamromulan** [comment]
Yeah.....This will be visited next most likely. Cleaning the whole github account. Sorry been way busy ;p 
Check out the new wiki instead. Lets build a knowledgebase and forum in a single repo :)
